Cython/Compiler/PyrexTypes.py

```diff
@@ -1556,7 +1556,14 @@ def attributes_known(self):
             self.scope = scope = Symtab.CClassScope('', None, visibility="extern")
             scope.parent_type = self
             scope.directives = {}
-            scope.declare_var("shape", CPtrType(c_long_type), None, cname="_shape", is_cdef=True)
+            shape = scope.declare_cfunction(
+                    "shape",
+                    CFuncType(CPtrType(c_long_type), [CFuncTypeArg("self", self, None)], nogil=True),
+                    pos=None,
+                    defining=0,
+                cname=" ")
+            shape.is_cgetter = 1
```

Comment:
mmmh, I really wanted to have something alike a property, what makes you think it's better to have a declare_cfunction?

Reply:
That's fine. What I meant was that calling `declare_cfunction()` here in order to get such a property seems like a leaky abstraction. It would be nicer if there was a method `declare_cgetter()` to call in places like this, which could then do the right setup internally, such as calling `declare_cfunction()`, but without requiring the outside code to know about these details.